lib/netsnmp/message.rb

```diff
@@ -130,10 +130,15 @@ def encode(pdu, security_parameters:, engine_boots: 0, engine_time: 0)
 
     # https://datatracker.ietf.org/doc/html/rfc7860#section-4.2.2 part 3
     # https://datatracker.ietf.org/doc/html/rfc3414#section-6.3.2 part 3
-    def authnone(auth_protocol)
+    def authnone(auth_protocol, require_authentication = true)
```

Comment:
Instead of a new parameter in the signature,  wouldn't it be better to just use the "security_level", which is 0 for engine probes?